### PR TITLE
add codeql

### PR DESCRIPTION
**Pull request by dorschw**

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+  schedule:
+    - cron: "1 14 * * 4"
```
Comment (tkatzir) [inline .github/workflows/codeql.yml]: Every Thursday at 14:01? Why right before the weekend?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+  schedule:
+    - cron: "1 14 * * 4"
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
```
Comment (tkatzir) [inline .github/workflows/codeql.yml:14]: Do we **need** ubuntu?

**Review by tkatzir — COMMENTED** [review]
Great!

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (dorschw) [reply .github/workflows/codeql.yml:14]: That's the suggested default. I don't mind using another.

Comment (tkatzir) [reply .github/workflows/codeql.yml:14]: Try alpine or debian.

**Pull request closed without merging**